kotlin-samples/corda5-negotiation/workflows/src/main/kotlin/com/r3/developers/samples/negotiation/workflows/Propose/ProposalFlowRequest.kt

```diff
@@ -0,0 +1,93 @@
+package com.r3.developers.samples.negotiation.workflows.Propose
+
+import com.r3.developers.samples.negotiation.Proposal
+import com.r3.developers.samples.negotiation.ProposalAndTradeContract.Propose
+import com.r3.developers.samples.negotiation.util.Member
+import com.r3.developers.samples.negotiation.workflows.util.FinalizeFlow
+import net.corda.v5.application.flows.*
+import net.corda.v5.application.marshalling.JsonMarshallingService
+import net.corda.v5.application.membership.MemberLookup
+import net.corda.v5.application.messaging.FlowMessaging
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.base.exceptions.CordaRuntimeException
+import net.corda.v5.base.types.MemberX500Name
+import net.corda.v5.ledger.common.NotaryLookup
+import net.corda.v5.ledger.utxo.UtxoLedgerService
+import java.time.Duration
+import java.time.Instant
+import java.util.*
+import java.util.List
+
+@InitiatingFlow(protocol = "proposal")
+class ProposalFlowRequest : ClientStartableFlow {
+    @CordaInject
+    var flowMessaging: FlowMessaging? = null
+
+    @CordaInject
+    var jsonMarshallingService: JsonMarshallingService? = null
+
+    @CordaInject
+    var memberLookup: MemberLookup? = null
+
+    @CordaInject
+    var notaryLookup: NotaryLookup? = null
+
+    @CordaInject
+    var utxoLedgerService: UtxoLedgerService? = null
+
+    @CordaInject
+    var flowEngine: FlowEngine? = null
+
+    @Suspendable
+    override fun call(requestBody: ClientRequestBody): String {
```

Comment:
Why does ProposalFlow return a string? Is it not supposed to return a UUID? Taking into account that Accept and Modify flows seem to be expecting the proposal ID to be a UUID